clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -592,6 +627,21 @@ class ScalarExprEmitter : public StmtVisitor<ScalarExprEmitter, mlir::Value> {
   buildCompoundAssign(const CompoundAssignOperator *E,
                       mlir::Value (ScalarExprEmitter::*F)(const BinOpInfo &));
 
+  QualType getPromotionType(QualType Ty) {
```

Comment:
Can you please add a comment to this function saying "This function is a good candidate to be part of a common AST helper between CIR and LLVM codegen"?

Reply:
Added

```
// TODO(cir): Candidate to be in a common AST helper between CIR and LLVM codegen.
```